apps/desktop/desktop_native/autotype/src/windows.rs

```diff
@@ -56,6 +60,44 @@ pub fn type_input(input: Vec<u16>) -> Result<(), ()> {
     send_input(keyboard_inputs)
 }
 
+/// Converts a valid shortcut key to an "up" keyboard input.
+///
+/// `input` must be a valid shortcut key: Control, Alt, Super, Shift, letters [a-z][A-Z]
+fn convert_shortcut_key_to_up_input(key: String) -> Result<INPUT, ()> {
+    const SHIFT_KEY: u8 = 0x10;
+    const SHIFT_KEY_STR: &str = "Shift";
+    const CONTROL_KEY: u8 = 0x11;
+    const CONTROL_KEY_STR: &str = "Control";
+    const ALT_KEY: u8 = 0x12;
+    const ALT_KEY_STR: &str = "Alt";
+    const LEFT_WINDOWS_KEY: u8 = 0x5B;
+    const LEFT_WINDOWS_KEY_STR: &str = "Super";
+
+    Ok(match key.as_str() {
+        SHIFT_KEY_STR => build_virtual_key_input(InputKeyPress::Up, SHIFT_KEY),
+        CONTROL_KEY_STR => build_virtual_key_input(InputKeyPress::Up, CONTROL_KEY),
+        ALT_KEY_STR => build_virtual_key_input(InputKeyPress::Up, ALT_KEY),
+        LEFT_WINDOWS_KEY_STR => build_virtual_key_input(InputKeyPress::Up, LEFT_WINDOWS_KEY),
+        _ => build_unicode_input(InputKeyPress::Up, encode_utf16_alphabetic_hotkey(key)?),
+    })
+}
+
+/// Given a letter that is a String, get the utf16 encoded
+/// decimal version of the letter as long as it meets the
+/// [a-z][A-Z] restriction.
```

Comment:
Can we add a comment here about why we don't need to explicitly encode to ut16, something like


```suggestion
/// Given a letter that is a String, get the utf16 encoded
/// decimal version of the letter as long as it meets the
/// [a-z][A-Z] restriction.
///
/// Because we only accept [a-z][A-Z], the decimal u16
/// cast of the letter is safe because the unicode code point
/// of these characters fits in a u16.
```

Reply:
Resolved in: https://github.com/bitwarden/clients/pull/16613/commits/8a17ea1eb126a335d83471ca26cd1669de7770f8